src/Storages/ObjectStorageQueue/ObjectStorageQueueMetadata.cpp

```diff
@@ -156,6 +156,10 @@ ObjectStorageQueueMetadata::~ObjectStorageQueueMetadata()
 
 void ObjectStorageQueueMetadata::startup()
 {
+    if (startup_called)
```

Comment:
This will not help without a mutex, I saw this crash you were fixing and did not make a fix, because a mutex is added in this PR already https://github.com/ClickHouse/ClickHouse/pull/77878

Reply:
Indeed, nice catch. OK, let's continue in https://github.com/ClickHouse/ClickHouse/pull/77878/files#r2009968810